java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java

```diff
@@ -132,6 +132,11 @@ public String sanitize(@NonNull URL url, @NonNull String jsonResponse) {
                     .delete(document, jsonConfiguration);
             }
 
+            //TODO: error handling within the map functions. any exceptions thrown within the map
+            //      function seem to be suppressed, and an empty [] left as the 'document'.
+            // ideas:
+            // jsonConfiguration.addEvaluationListeners(); -->
```

Comment:
Not sure, I mean it could work but at the moment I don't see an improvement in comparison with current solution and using evaluation listeners. At the end, `EvaluationListener` will be like this (some dirty code):

```
static class X implements EvaluationListener {

    @Override
    public EvaluationContinuation resultFound(FoundResult found) {
        if (found.result() == null) {
            return EvaluationContinuation.CONTINUE;
        }
        if (Objects.equals(found.path(), "my-path")) {
            return EvaluationContinuation.CONTINUE;
        }
        
        return EvaluationContinuation.ABORT;
    }
}
```